src/lib/workRelationsWorkTypeFactoryUtils.js

```diff
@@ -0,0 +1,160 @@
+  return manifestations.filter((manifestation) =>
+    [RelationTypeEnum.ISADAPTATIONOF.key].includes(manifestation?.relationType)
+  );
+}
+
+export function enrichDebateArticle(manifestations) {
+  return manifestations.filter((manifestation) =>
+    [RelationTypeEnum.DISCUSSES.key, RelationTypeEnum.DISCUSSEDIN.key].includes(
+      manifestation?.relationType
+    )
+  );
+}
+
+export function enrichLiterature(manifestations) {
+  return manifestations.filter((manifestation) =>
+    [RelationTypeEnum.HASADAPTATION.key].includes(manifestation?.relationType)
+  );
+}
+
+export function mapRelationTypes(manifestation) {
+  return RelationTypeEnum[manifestation?.relationType?.toUpperCase()]?.workType;
+}
+
+export function enrichBySpecificWorkType(manifestations, work) {
+  const relationTypes = uniq(manifestations.map(mapRelationTypes));
+
+  const enrichMapper = {
+    [WorkTypeEnum.ARTICLE]: () => enrichArticleSeries(manifestations, work),
+    [WorkTypeEnum.MOVIE]: () => enrichMovie(manifestations),
+    [WorkTypeEnum.DEBATEARTICLE]: () =>
+      enrichDebateArticle(manifestations, work),
+    [WorkTypeEnum.LITERATURE]: () => enrichLiterature(manifestations, work),
+  };
+
+  return relationTypes
+    .map((relationType) => enrichMapper[relationType]?.())
+    .filter(
+      (manifestation) => manifestation && typeof manifestation !== "undefined"
+    )
+    .flat();
+}
+
+export function filterFieldsInElement(entry) {
+  delete entry.ownerWork;
+  delete entry.relations;
+
+  return entry;
+}
+
+export function parseSingleRelation(manifestation, relationType) {
+  return {
+    ...manifestation,
+    ...(relationType && { relationType: relationType }),
+    ...(manifestation?.ownerWork?.workId && {
+      workId: manifestation?.ownerWork?.workId,
+    }),
+  };
+}
+
+export function parseSingleManifestation(
+  relationTypeArray,
+  passedFunction = parseSingleRelation
+) {
+  return (
+    relationTypeArray?.[1]?.map((manifestation) =>
+      passedFunction(manifestation, relationTypeArray[0])
+    ) || []
+  );
+}
+
+function getAllWorksWithRelationTypeAndWorkId(work) {
+  return (
+    (work?.relations &&
+      Object.entries(work?.relations)
+        ?.map((relationTypeArray) =>
+          parseSingleManifestation(relationTypeArray)
+        )
+        .flat()) ||
+    []
+  );
+}
+
+function getUniqWorkWithWorkId(manifestations) {
+  return uniqWith(manifestations, (a, b) => isEqual(a.workId, b.workId));
+}
+
+export function parseRelations(work) {
+  return pipe(work, [
+    getAllWorksWithRelationTypeAndWorkId,
+    getUniqWorkWithWorkId,
+    (manifestations) => enrichBySpecificWorkType(manifestations, work),
+    groupManifestations,
+    flattenGroupedSortedManifestations,
+    (manifestations) => manifestations.map(filterFieldsInElement),
+  ]);
+}
+
+export function workRelationsWorkTypeFactory(work) {
+  const parsedRelations = work && parseRelations(work);
+
+  return {
+    flatRelations: parsedRelations,
+  };
+}
```

Comment:
in general - comments on methods - we need to know what  and why and return types and parameters ... on so on